sklearn/tree/tree.py

```diff
@@ -655,6 +655,8 @@ class DecisionTreeClassifier(ClassifierMixin, BaseDecisionTree):
         If None, the random number generator is the RandomState instance used
         by `np.random`.
 
+        Note: The source of randomness is exlained in the notes.
```

Comment:
I think we should remove the details above ("if int...") and instead link to the glossary, as illustrated in https://github.com/scikit-learn/scikit-learn/issues/15222.

And we could just move what's in the notes here.

```
random_state : int, np.random.RandomStateInstance or None, default=None
	Controls the randomness of the estimator. [notes section here]. See 
	term:`random_state` for details.
```